src/sprites/sprite.js

```diff
@@ -140,11 +149,41 @@ class Sprite {
         const blocksContainer = this.blocks._blocks;
         const originalBlocks = Object.keys(blocksContainer).map(key => blocksContainer[key]);
         const copiedBlocks = JSON.parse(JSON.stringify(originalBlocks));
-        newBlockIds(copiedBlocks);
+        const oldToNew = newBlockIds(copiedBlocks);
         copiedBlocks.forEach(block => {
             newSprite.blocks.createBlock(block);
         });
 
+        Object.keys(this.comments).forEach(commentId => {
+            const comment = this.comments[commentId];
+            const newComment = new Comment(
+                null, // use new comment id
+                comment.text,
+                comment.x,
+                comment.y,
+                comment.width,
+                comment.height,
+                comment.minimized
+            );
+            if (comment.blockId) {
+                const newBlockId = oldToNew[comment.blockId];
+                if (newBlockId) {
+                    newComment.blockId = newBlockId;
+                    const newBlock = newSprite.blocks.getBlock(newBlockId);
+                    if (newBlock) {
+                        newBlock.comment = newComment.id;
+                    } else {
+                        log.warn(`Could not find block with id ${newBlockId
+                        } associated with comment ${newComment.id}`);
+                    }
+                } else {
+                    log.warn(`Could not find block with id ${comment.blockId
+                    } associated with comment ${comment.id} in the block ID mapping`);
```

Comment:
Will either this code path or the one above be hit because of #2501, or is this a "just in case" thing?

Reply:
It's probably "just in case" thing - deserialization should fix 2501 case

Reply:
But because we removed deserialization changes from this PR, this handles 2501 case now.